src/scene.rs

```diff
@@ -28,4 +29,115 @@ pub struct Scene {
     /// See [`Background`](struct.Background.html).
     pub background: Background,
 }
+
+/// `SyncGuard` is used to obtain information about in scene nodes in most effective way.
+///
+/// # Examples
+///
+/// Imagine that you have your own helper type `Enemy`:
+///
+/// ```rust
+/// # #[macro_use]
+/// # extern crate three;
+/// struct Enemy {
+///     mesh: three::Mesh,
+///     is_visible: bool,
+/// }
+/// three_object_wrapper!(Enemy::mesh);
+/// # fn main() {}
+/// ```
+///
+/// You need this wrapper around `three::Mesh` to cache some information - in our case, visibility.
+///
+/// In your game you contain all your enemy objects in `Vec<Enemy>`. In the main loop you need
+/// to iterate over all the enemies and make them visible or not, basing on current position.
+/// The most obvious way is to use [`Object::sync`], but it's not the best idea from the side of
+/// performance. Istead, you can create `SyncGuard` and use its `update` method to effectively
+/// walk through every enemy in your game:
+///
+/// ```rust,no_run
+/// # #[macro_use]
+/// # extern crate three;
+/// # #[derive(Clone)]
+/// # struct Enemy {
+/// #     mesh: three::Mesh,
+/// #     is_visible: bool,
+/// # }
+/// # three_object_wrapper!(Enemy::mesh);
+/// # fn main() {
+/// # let mut win = three::Window::new("SyncGuard example");
+/// # let geometry = three::Geometry::default();
+/// # let material = three::material::Basic { color: three::color::RED, map: None };
+/// # let mesh = win.factory.mesh(geometry, material);
+/// # let mut enemy = Enemy { mesh, is_visible: true };
+/// # enemy.set_parent(&win.scene);
+/// # let mut enemies = vec![enemy];
+/// # while true {
+/// let mut sync = win.scene.with_sync();
+/// sync.update(enemies.iter_mut(), |enemy, node| {
+///     let position = node.world_transform.position;
+///     if position.x > 10.0 {
+///         enemy.is_visible = false;
+///         enemy.set_visible(false);
+///     } else {
+///         enemy.is_visible = true;
+///         enemy.set_visible(true);
+///     }
+/// });
+/// # }}
+/// ```
+///
+/// [`Object::sync`]: ../struct.Object.html#method.sync
+pub struct SyncGuard<'a> {
+    pub(crate) scene: &'a mut Scene,
+    pub(crate) scene_id: Option<Uid>,
+}
+
+impl<'a> SyncGuard<'a> {
+    /// Walk through every item in `iterator`, obtain its [`Node`] and call user-supplied closure.
+    ///
+    /// # Panics
+    /// Panics if `scene` doesn't have this `Object`.
+    ///
+    /// [`Node`]: ../node/struct.Node.html
+    pub fn update<T: 'a, I, F>(
+        &mut self,
+        iterator: I,
```

Comment:
In future, it might be beneficial to have an `Extend` implementation, so the user can pass the guard around and iterate over different types.

Reply:
Mm... I'm not quite following how `Extend` can be be useful

Reply:
The idea I had in mind was to make `SyncGuard` like a collection of objects to be synchronised. Perhaps `Extend` isn't a great way to do this. This is what I had in mind:

```rust
let mut sharks = vec![Shark];
let mut lasers = vec![Laser];
{
    scene.sync_guard()
        .update(&mut sharks, |shark, node| shark.attach_to_head(Laser))
        .update(&mut laser, |laser, node| laser.fire(node.transform))
        .exec();
}
```

Reply:
I don't think it's better than current design proposal. You can do the same thing with multiple `update`s, and without any performance issues.